app/lib/features/chat/pages/room_page.dart

```diff
@@ -37,8 +33,11 @@ class _RoomPageConsumerState extends ConsumerState<RoomPage> {
   void initState() {
     super.initState();
     WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
-      ref.watch(chatRoomProvider.notifier).init(widget.convo.getRoomIdStr());
-      ref.watch(chatRoomProvider.notifier).fetchUserProfiles();
+      final convo = ref.watch(currentConvoProvider);
+      if (convo != null) {
+        ref.watch(chatRoomProvider.notifier).init(convo.getRoomIdStr());
+        ref.watch(chatRoomProvider.notifier).fetchUserProfiles();
```

Comment:
this feels kinda wrong here... like ... it already felt wrong before, but not even more so.

Shouldn't that be happening inside the `chatRoomProvider` listening to the `currentConvoProvider`? Rather than being initiated by a widget? If this is too much work to do now, I am okay with keeping it as-is... but I think generally, we rather want the providers coupling to happen at the provider-side rather than at the widget-level, don't you think?

Reply:
Hmm yes, these events should happen on business logic layer and shouldn't bloat up on the interface. I will try to put these where intended.